src/auth/externalclient.ts

```diff
@@ -29,10 +29,15 @@ import {
   IdentityPoolClientOptions,
 } from './identitypoolclient';
 import {AwsClient, AwsClientOptions} from './awsclient';
+import {
+  PluggableAuthClient,
+  PluggableAuthClientOptions,
+} from './pluggableauthclient';
 
 export type ExternalAccountClientOptions =
   | IdentityPoolClientOptions
-  | AwsClientOptions;
+  | AwsClientOptions
+  | PluggableAuthClientOptions;
```

Comment:
I believe this type change will mean we should treat the change as breaking, we can link against `googleapis-common` and `google-gax` to confirm.

IMO this isn't a big deal, we just take a major bump for the auth client, which allows us to intentionally upgrade auth in `google-gax`.

Reply:
That seems fine to me, the only change I'll have to make for that is to add the "next major: breaking change" tag to the final PR into main correct?

Reply:
I would merge this PR as:

```
feat!: Adding Plugable Auth Support for ADC
```

And we should make sure to remember this is a breaking change when we land the feature -- before we merge your branch I would also like to test with gax, grpc, google-api-nodejs-client, and a GAPIC library.